.github/workflows/test-e2e.yaml

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The test-e2e action does not check out the repo. Instead, it triggers the e2e
+# Cloud Build using this action's commit SHA, then waits for the cloud build
+# run to complete.
+#
+# This action uses a Cloud Build trigger rather than `gcloud build submit`
+# because we want Cloud Build to check out the repo. We don't want to upload
+# the source from this github action.
+#
+# We don't want e2e build log to be printed in this Github Action's log
+# because it may contain secrets.
+#
+name: e2e-test-trigger
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
+jobs:
+  build:
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'e2e: run' }}"
```

Comment:
nit: for consistency should we have the label called `tests: run`?

Reply:
I want to reserve `tests: run` to run just the the "build and test" action, and have "e2e: run" for this action. Remember, that if two jobs launch at the same moment, one of them will fail due to GitHub rate limits.

Reply:
do you have an example of where this has happened or the error message resulting from rate limits?

Reply:
Is there documentation somewhere about these Github rate limits you can point me to?